packages/shared/src/hooks/useInputFieldFunctions.ts

```diff
@@ -43,6 +43,16 @@ function useInputFieldFunctions<
     setValidInput(inputRef.current.checkValidity());
   }, 1500);
 
+  useEffect(() => {
+    if (inputRef.current?.value) {
+      setInputLength(inputRef.current.value.length);
+      const inputValidity = inputRef.current.checkValidity();
+      if (inputValidity) {
+        setValidInput(true);
+      }
+    }
+  }, [inputRef]);
```

Comment:
Hmm how does this work, because `inputRef` is not reactive or state so reference is always the same? We should instead check this in `setInput` I think.

Reply:
I want it to be stale, don't want it to trigger all the time.
It basically waits till it exists and then sets the starting point once, otherwise we already handle on input change.

Reply:
So this should trigger once when input/hook renders? Wonder why it was not needed before?

Reply:
![Screenshot 2023-10-06 at 13 38 11](https://github.com/dailydotdev/apps/assets/554874/195150bf-9676-4fae-94de-a7dfe148d476)
 
To give more context, the above (right 0) would show 250 before this fix.
I think we just never noticed it.
But basically it would never set anything if the value would be set already on load.

> Note: it already worked if you removed a char and re-typed though

Reply:
Makes sense, it just looked weird and harder to understand. Hoping down the line we can move to some input library like `react-hook-form`. 🤞

Reply:
Yeah, more and more edge-cases and duct-tape.
Lol really be nice to see a uniform library